packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,293 @@
+	readonly blockHash: HexString32Bytes | null;
+	readonly blockNumber: Uint | null;
+	readonly from: Address;
+	readonly hash: HexString32Bytes;
+	readonly transactionIndex: Uint | null;
+};
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.json#L24
+export type TransactionWithSender = TransactionUnsigned & { from: Address };
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/block.json#L2
+export interface Block {
+	readonly parentHash: HexString32Bytes;
+	readonly sha3Uncles: HexString32Bytes;
+	readonly miner: HexString;
+	readonly stateRoot: HexString32Bytes;
+	readonly transactionsRoot: HexString32Bytes;
+	readonly receiptsRoot: HexString32Bytes;
+	readonly logsBloom: HexString256Bytes | null;
+	readonly difficulty?: Uint;
+	readonly number: Uint | null;
+	readonly gasLimit: Uint;
+	readonly gasUsed: Uint;
+	readonly timestamp: Uint;
+	readonly extraData: HexStringBytes;
+	readonly mixHash: HexString32Bytes;
+	readonly nonce: HexStringBytes | null;
+	readonly totalDifficulty: Uint;
+	readonly baseFeePerGas?: Uint;
+	readonly size: Uint;
+	readonly transactions: TransactionHash[] | TransactionSigned[];
+	readonly uncles: Uncles;
+	readonly hash: HexString32Bytes | null;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L59
+export type Topic = HexString256Bytes;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L2
+export interface Log {
+	readonly removed?: boolean;
+	readonly logIndex?: Uint;
+	readonly transactionIndex?: Uint;
+	readonly transactionHash?: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly address?: Address;
+	readonly data?: HexStringBytes;
+	readonly topics?: Topic[];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L44
+export interface ReceiptInfo {
+	readonly transactionHash: HexString32Bytes;
+	readonly transactionIndex: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly from?: Address;
+	readonly to?: Address;
+	readonly cumulativeGasUsed?: Uint;
+	readonly gasUsed?: Uint;
+	readonly contractAddress?: Address;
+	readonly logs?: Log[];
+	readonly logsBloom?: HexString256Bytes;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/client.json#L2
+export type SyncingStatus =
+	| { startingBlock: Uint; currentBlock: Uint; highestBlock: Uint }
+	| boolean;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/eth/fee_market.json#L53
+export interface FeeHistoryResult {
+	readonly oldestBlock: Uint;
+	readonly baseFeePerGas: Uint;
+	readonly reward: number[][];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L28
+export interface Filter {
+	readonly fromBlock?: Uint;
+	readonly toBlock?: Uint;
+	readonly address?: Address | Address[];
+	readonly topics?: Topic[];
```

Comment:
```suggestion
	readonly topics?: Topic | Topic[] | null [];
```

Example from legacy spec:

```bash
"topics": ["0x000000000000000000000000a94f5374fce5edbc8e2a8697c15331677e6ebf0b", null, ["0x000000000000000000000000a94f5374fce5edbc8e2a8697c15331677e6ebf0b", "0x0000000000000000000000000aff3454fce5edbc8cca8697c15331677e6ebccc"]]
```

Created an [issue](https://github.com/ethereum/execution-apis/issues/118) for this